tests/lib/cli.js

```diff
@@ -803,6 +803,29 @@ describe("cli", () => {
         });
     });
 
+    describe("when given the exit-on-fatal-error flag", () => {
+        it("should not change exit code if no fatal errors are reported", async () => {
+            const filePath = getFixturePath("exit-on-fatal-error/no-fatal-error.js");
+            const exitCode = await cli.execute(`--no-ignore --exit-on-fatal-error ${filePath}`);
+
+            assert.strictEqual(exitCode, 0);
+        });
+
```

Comment:
It would be good to add a test with a linting error that isn't fatal (exit code 1 with `--exit-on-fatal-error`).